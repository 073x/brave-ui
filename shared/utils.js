const tokens = require("./tokens");

const pathRegex = /([a-zA-Z]+)(?:\[|\.)([a-zA-Z0-9-]+)\]?/
const token = (path) => {
    let properties;
    if (path.includes(".") || path.includes("[")) {
        properties = path.match(pathRegex).slice(1);
    } else {
        properties = [path];
    }

    let value = tokens;
    for (prop of properties) {
        value = value[prop];
    }

    if (!value) {
        throw new Error(`${path} does not exist.`);
    }

    return value;
}

module.exports = {
    token
}