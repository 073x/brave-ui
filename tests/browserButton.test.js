/* global jest, expect, describe, it, afterEach */
import React from 'react'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import {resumeStyleInjection} from './lib/utils'
import BrowserButton from '../brave-ui/src/browserButton'

describe('browserButton tests', () => {
  afterEach(() => {
    // Note: this is required for every test file
    // see utils.resumeStyleInjection for more info
    resumeStyleInjection()
  })

  it('matches the snapshot', () => {
    const component = <BrowserButton label='browserButton as default' />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component', () => {
    const wrapper = shallow(<BrowserButton />)
    const assertion = wrapper.find('[data-test-id="browserButton"]').length
    expect(assertion).toBe(1)
  })

  it('defines a custom size', () => {
    const wrapper = shallow(<BrowserButton size='13px' />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(expect.objectContaining({'--size': '13px'}))
  })

  it('defines a custom font size', () => {
    const wrapper = shallow(<BrowserButton fontSize='60px' />)
    const assertion = wrapper.find('button').props().style
    expect(assertion).toEqual(expect.objectContaining({'--fontSize': '60px'}))
  })

  it('can pass locale', () => {
    const wrapper = shallow(<BrowserButton l10nId='NESPRESSO' />)
    const assertion = wrapper.find('button').props()['data-l10n-id']
    expect(assertion).toBe('NESPRESSO')
  })

  it('can be disabled', () => {
    const wrapper = shallow(<BrowserButton disabled />)
    const assertion = wrapper.find('button').props().disabled
    expect(assertion).toBe(true)
  })

  it('defaults to the default button', () => {
    const wrapper = shallow(<BrowserButton />)
    const assertion = wrapper.find('button').props()['data-test-as']
    expect(assertion).toBe('default')
  })

  it('can be set as primary', () => {
    const wrapper = shallow(<BrowserButton as='primary' />)
    const assertion = wrapper.find('button').props()['data-test-as']
    expect(assertion).toBe('primary')
  })

  it('can be set as secondary', () => {
    const wrapper = shallow(<BrowserButton as='secondary' />)
    const assertion = wrapper.find('button').props()['data-test-as']
    expect(assertion).toBe('secondary')
  })

  it('can have a custom label', () => {
    const wrapper = shallow(<BrowserButton label='CLOONEY' />)
    const assertion = wrapper.find('button').text()
    expect(assertion).toBe('CLOONEY')
  })

  it('can respond to click', () => {
    const value = 'FANCY BUTTON CLICKED'
    const onClick = jest.fn()
    const wrapper = shallow(<BrowserButton onClick={onClick} />)
    wrapper.find('button').simulate('click', value)
    expect(onClick).toBeCalledWith(value)
  })
})
