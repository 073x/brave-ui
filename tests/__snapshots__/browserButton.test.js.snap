// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`browserButton tests matches the snapshot 1`] = `
<button
  className="browserButton_1b3ew9l"
  data-l10n-id={undefined}
  data-test-as="default"
  data-test-id="browserButton"
  disabled={undefined}
  onClick={undefined}
  style={
    Object {
      "--bg": "#fff",
      "--color": "#5a5a5a",
      "--hoverColor": "#000",
      "--primary-bg": "linear-gradient(#ff7a1d, #ff5000)",
      "--primary-borderHoverColor": "#fff",
      "--primary-color": "#fff",
      "--primary-gradient1": "#ff7a1d",
      "--primary-gradient2": "#ff5000",
      "--primary-hoverColor": "#fff",
      "--secondary-bg": "linear-gradient(#fff, #ececec)",
      "--secondary-borderHoverColor": "#444",
      "--secondary-color": "#666",
      "--secondary-hoverColor": "rgb(153, 153, 153)",
    }
  }
>
  browserButton as default
</button>
`;
